tests/unit/test_download.py

```diff
@@ -325,3 +325,18 @@ def test_insecure_host_cache_is_not_enabled(self, tmpdir):
         )
 
         assert not hasattr(session.adapters["https://example.com/"], "cache")
+
+
+def test_parse_credentials():
+    auth = MultiDomainBasicAuth()
+    assert auth.parse_credentials(u"foo:bar@example.com") == (u'foo', u'bar')
```

Comment:
OK. We can change it to use str in Python 2. I don't know if the author of this patch would be interested to work on this, because the original patch is from Nov '15. If not, I volunteer to create a new pull request with these changes, if that's fine with you.